presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -525,6 +526,11 @@ public SystemSessionProperties(
                         "Experimental: Enable statistics calculator",
                         featuresConfig.isEnableStatsCalculator(),
                         false),
+                integerProperty(
+                        MAX_TASKS_PER_STAGE,
+                        "Maximum number of tasks for a non source distributed stage",
```

Comment:
`for non-source stages` OR `for intermediate stages` ?

Reply:
@nezihyigitbasi : Thanks for the comment! In our case it's more about whether the stage/fragment's distribution is `SOURCE_DISTRIBUTION` (https://github.com/prestodb/presto/blob/941255bcab3cda4167fe589f3df468dc989f68c7/presto-main/src/main/java/com/facebook/presto/sql/planner/SystemPartitioningHandle.java#L63) . For example, a collocated join might be a leaf stage, but the distribution of that stage/fragment would follow table partitioning. 

Maybe we can say `Maximum number of tasks for stage, unless its partitioning handle is SOURCE_DISTRIBUTION`  to avoid confusion?

Reply:
I'm not sure if an average end user would now what it is a partitioning handle.

Reply:
`Maximum number of tasks for a non source distributed stage` - I think it should be pretty intuitive.

The user should understand that it is not possible (at least for all the cases) to limit the number of source distributed tasks. As the source distribute tasks might be bound to the data location. So, somehow it makes sense. However i agree that it is still quite confusing.